src/system/SystemPacketBuffer.cpp

```diff
@@ -355,6 +357,29 @@ PacketBuffer * PacketBuffer::Consume(uint16_t aConsumeLength)
     return lPacket;
 }
 
+CHIP_ERROR PacketBuffer::Read(uint8_t * aDestination, size_t aLength)
+{
+    PacketBuffer * lPacket = this;
+
+    if (aLength > TotalLength())
+    {
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    while (lPacket != nullptr && aLength > 0)
```

Comment:
How can `lPacket` become null here, as long as our `TotalLength` is not busted?

Might be worth a comment explaining why the check is here, even if it's just a belt-and-suspenders against a corrupted `TotalLength` or something....  But then we'd want to return an error if we do run off the end of the buffer chain before aLength is 0, I'd think.  Otherwise the caller thinks we gave then N bytes of data but we gave them less than N and they have uninitialized data.

Reply:
I think it's because of unconditional `lPacket = lPacket->ChainedBuffer();` at the bottom of the loop.